doc/source/migrations/doc_changelog_setup.rst

```diff
@@ -0,0 +1,177 @@
+.. _doc_changelog_action_setup:
+
+Doc-changelog action setup
+==========================
+
+To set up your repository to use ``ansys/actions/doc-changelog``, see the
+`changelog implementation in PyAnsys-Geometry <https://github.com/ansys/pyansys-geometry/pull/1023/files>`_
+or follow these steps:
+
+
+1. Add the following lines to the pyproject.toml file, replacing ``{repo-name}`` with the name of the repository. For example, ``pyansys-geometry``.
+
+.. code:: python
+
+    [tool.towncrier]
+    directory = "doc/changelog.d"
+    filename = "CHANGELOG.md"
+    start_string = "<!-- towncrier release notes start -->\n"
+    underlines = ["", "", ""]
+    template = "doc/changelog.d/changelog_template.jinja"
+    title_format = "## [{version}](https://github.com/ansys/{repo-name}/releases/tag/v{version}) - {project_date}"
+    issue_format = "[#{issue}](https://github.com/ansys/{repo-name}/pull/{issue})"
+
+    [[tool.towncrier.type]]
+    directory = "added"
+    name = "Added"
+    showcontent = true
+
+    [[tool.towncrier.type]]
+    directory = "changed"
+    name = "Changed"
+    showcontent = true
+
+    [[tool.towncrier.type]]
+    directory = "fixed"
+    name = "Fixed"
+    showcontent = true
+
+    [[tool.towncrier.type]]
+    directory = "dependencies"
+    name = "Dependencies"
+    showcontent = true
+
+    [[tool.towncrier.type]]
+    directory = "miscellaneous"
+    name = "Miscellaneous"
+    showcontent = true
+
+|
+
+2. Create the ``doc/changelog.d`` folders, and add a file within ``changelog.d`` named ``changelog_template.jinja`` containing the following lines:
+
+.. code:: jinja
+
+    {% if sections[""] %}
+    {% for category, val in definitions.items() if category in sections[""] %}
+
+    ### {{ definitions[category]['name'] }}
+
+    {% for text, values in sections[""][category].items() %}
+    - {{ text }} {{ values|join(', ') }}
+    {% endfor %}
+
+    {% endfor %}
+    {% else %}
+    No significant changes.
+
+
+    {% endif %}
+
+|
+
+3. Add the following lines to the ``CHANGELOG.md`` file, replacing ``{repo-name}`` with the name of the repository:
+
+.. code:: md
+
+    This project uses [towncrier](https://towncrier.readthedocs.io/) and the changes for the upcoming release can be found in <https://github.com/ansys/{repo-name}/tree/main/docs/changelog.d/>.
+
+    <!-- towncrier release notes start -->
+
+
+.. note::
+
+    If ``CHANGELOG.md`` already has sections for previous releases, make sure to put the
+    ``"towncrier release notes start"`` comment preceding the release sections. For example:
+
+    .. code:: md
+
+        <!-- towncrier release notes start -->
+
+        ## [0.10.7](https://github.com/ansys/pymechanical/releases/tag/v0.10.7) - February 13 2024
+
+|
+
+4. Update ``.github/workflows/label.yml`` to use the changelog action.
+
+Change the ``pull_request`` trigger at the top of the preceding ``.yml`` file, so it lists the pull request actions that cause the workflows to run.
```

Comment:
```suggestion
Change the ``pull_request`` trigger at the top of the preceding ``.yml`` file so that it lists the pull request actions that cause the workflows to run:
```